BookingSystem/Frontend/src/shared/utils/optionsService.js

```diff
@@ -0,0 +1,50 @@
+class OptionsService {
+  static getEmptyOption() {
+    return [
+      {
+        value: "",
+        label: ""
+      }
+    ];
+  }
+
+  static getOptions(options, labelProperty, valueProperty = null) {
+    if (!valueProperty) {
+      valueProperty = labelProperty;
+    }
+
+    let selectEntries = this.getEmptyOption();
+    const valueMap = [];
+    if (options) {
+      options.forEach(obj => {
+        const value = obj[valueProperty];
+        if (!valueMap[value]) {
+          valueMap[value] = true;
+          selectEntries.push({
+            value,
+            label: obj[labelProperty]
+          });
+        }
+      });
+    }
+
+    return selectEntries;
+  }
+
+  static getFilteredOptions(
+    options,
+    filterProp,
+    filterValue,
+    labelProp,
+    labelValue
+  ) {
+    if (!options) {
+      return [];
+    } else {
```

Comment:
this else doesn't make sense since you have "return" above

Reply:
Nope. That's for case when passed options empty. If no else block executes.